qa/rpc-tests/wallet_doublespend.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Zcash developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+import shutil
+import os.path
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    NU5_BRANCH_ID,
+    assert_equal,
+    get_coinbase_address,
+    nuparams,
+    start_nodes,
+    stop_nodes,
+    wait_bitcoinds,
+    wait_and_assert_operationid_status,
+)
+
+from decimal import Decimal
+
+# Test wallet behaviour with the Orchard protocol
+class WalletDoubleSpendTest(BitcoinTestFramework):
+    def __init__(self):
+        super().__init__()
+        self.num_nodes = 4
+
+    def setup_nodes(self):
+        return start_nodes(self.num_nodes, self.options.tmpdir, [[
+            nuparams(NU5_BRANCH_ID, 201),
+        ]] * self.num_nodes)
+
+    def run_test_for_recipient_type(self, recipient_type):
+        # Get a couple of new accounts
+        acct0a = self.nodes[0].z_getnewaccount()['account']
+        ua0a = self.nodes[0].z_getaddressforaccount(acct0a, [recipient_type])['address']
+
+        acct0b = self.nodes[0].z_getnewaccount()['account']
+        ua0b = self.nodes[0].z_getaddressforaccount(acct0b, [recipient_type])['address']
+
+        # Get a new UA for account 1
+        acct1 = self.nodes[1].z_getnewaccount()['account']
+        addrRes1 = self.nodes[1].z_getaddressforaccount(acct1, [recipient_type])
+        ua1 = addrRes1['address']
+
+        # Create an Orchard note on node 1
+        recipients = [{"address": ua1, "amount": Decimal('10')}]
+        myopid = self.nodes[0].z_sendmany(get_coinbase_address(self.nodes[0]), recipients, 1, 0, 'AllowRevealedSenders')
+        wait_and_assert_operationid_status(self.nodes[0], myopid)
+
+        self.sync_all()
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+        # Check the value sent to ua1 was received
+        assert_equal(
+                {'pools': {recipient_type: {'valueZat': Decimal('1000000000')}}, 'minimum_confirmations': 1},
```

Comment:
`Decimal` isn't needed for zatoshi-denominated values.
```suggestion
                {'pools': {recipient_type: {'valueZat': 1000000000}}, 'minimum_confirmations': 1},
```